src/ClimaAtmos/Dycore/src/VanillaAtmosDiscretizations_cuda.jl

```diff
@@ -17,7 +17,9 @@ function knl_volumegrad!(::Val{2}, ::Val{N}, ::Val{nmoist}, ::Val{ntrace},
   s_u = @cuStaticSharedMem(eltype(Q), (Nq, Nq))
   s_v = @cuStaticSharedMem(eltype(Q), (Nq, Nq))
   s_T = @cuStaticSharedMem(eltype(Q), (Nq, Nq))
-
+  q_m = @CuStaticSharedMem(eltype(Q),(3,))
```

Comment:
You do not want to use `CuStaticSharedMem` here but an `MArray`

Reply:
Noted - thanks : This is work in progress - the PR is tested against the CPU in branch moist-thermo; (I've reverted VanillaAtmosDiscretizations_cuda.jl in line with CLIMA-master in the meantime). This addresses your cuda kernel related comments for now.